python/ray/data/_internal/arrow_block.py

```diff
@@ -72,6 +72,20 @@ def get_concat_and_sort_transform(context: DataContext) -> Callable:
         return transform_pyarrow.concat_and_sort
 
 
+def is_nan(value):
+    return isinstance(value, float) and np.isnan(value)
+
+
+def keys_equal(keys1, keys2):
+    if len(keys1) != len(keys2):
+        return False
+    for k1, k2 in zip(keys1, keys2):
+        if (is_nan(k1) and is_nan(k2)) or k1 == k2:
```

Comment:
nit - can you flip the logic so that we shortcircuit

like if not X return False